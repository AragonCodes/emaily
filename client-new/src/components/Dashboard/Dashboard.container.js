import { connect } from 'react-redux';
import { surveyOperations } from '../../domain/survey';
import DashboardComponent from './Dashboard';

const mapStateToProps = ({ surveys }) => {
	return { surveys };
}

const mapDispatchToProps = (dispatch) => {
	const fetchSurveys = () => {
		dispatch(surveyOperations.fetchSurveys())
	};

	return { fetchSurveys };
};

const DashboardContainer = connect(
	mapStateToProps,
	mapDispatchToProps
)(DashboardComponent);

export default DashboardContainer;