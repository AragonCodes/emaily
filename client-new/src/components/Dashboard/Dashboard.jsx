import React, { useEffect } from 'react';
import { Link } from 'react-router-dom';

function Dashboard({ surveys, fetchSurveys }) {
	useEffect(() => {
		fetchSurveys();
	}, []);

	function renderSurveyList() {
		if (surveys.length === 0) return (<h1 className={'center'}>No Surveys Found</h1>);

		return (
			<div>
				{
					surveys
						.sort((a, b) => (new Date(b.dateSent) - new Date(a.dateSent)))
						.map(survey => (
							<div className="card darken-1" key={survey._id}>
								<div className="card-content">
									<span className="card-title">{survey.title}</span>
									<p>
										{survey.body}
									</p>
									<p className="right">
										Sent On: {new Date(survey.dateSent).toLocaleDateString()}
									</p>
								</div>
								<div className="card-action">
									<a>Yes: {survey.yes}</a>
									<a>No: {survey.no}</a>
								</div>
							</div>
						))
				}
			</div>
		);
	}

	return (
		<div>
			{renderSurveyList()}
			<div className="fixed-action-btn">
				<Link to="/surveys/new" className="btn-floating btn-large red">
					<i className="material-icons">add</i>
				</Link>
			</div>
		</div>
	);
};

export default Dashboard;
