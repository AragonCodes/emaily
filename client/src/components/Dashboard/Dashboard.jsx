import React, { useEffect } from 'react';
import { Link } from 'react-router-dom';

function Dashboard({ surveys, fetchSurveys }) {
	useEffect(() => {
		fetchSurveys();
	}, []);

	function renderSurveyList() {
		if (surveys.length === 0) return (
			<div className="center">
				<h1 className={'center'}>No surveys found</h1>
				<h4 style={{ margin: '2rem' }}>Start getting feedback form your usersâ€¦</h4>
				<Link to="/surveys/new" className="btn-large red accent-3">
					Create a new survey
			</Link>
			</div>
		);

		return (
			<div>
				{
					surveys
						.sort((a, b) => (new Date(b.dateSent) - new Date(a.dateSent)))
						.map(survey => (
							<div className="card darken-1" key={survey._id}>
								<div className="card-content">
									<span className="card-title">{survey.title}</span>
									<p>
										{survey.body}
									</p>
									<p className="right">
										Sent On: {new Date(survey.dateSent).toLocaleDateString()}
									</p>
								</div>
								<div className="card-action">
									<a className="red-text text-accent-2">Yes: {survey.yes}</a>
									<a className="red-text text-accent-2">No: {survey.no}</a>
								</div>
							</div>
						))
				}
			</div>
		);
	}

	return (
		<div>
			{renderSurveyList()}
			<div className="fixed-action-btn">
				<Link to="/surveys/new" className="btn-floating pulse btn-large red accent-2">
					<i className="material-icons">add</i>
				</Link>
			</div>
		</div>
	);
};

export default Dashboard;
