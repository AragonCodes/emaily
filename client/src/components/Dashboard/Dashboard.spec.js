import React from 'react';
import { shallow } from 'enzyme';
import Dashboard from './Dashboard';

describe('Dashboard', () => {
	it('should render correctly', () => {
		const wrapper = shallow(
			<Dashboard
				surveys={[
					{
						yes: 1,
						no: 0,
						_id: '5e3aff6d39558d00177e201a',
						title: 'A real campaign',
						subject: 'Customer Review',
						body: 'Did you like the service provided by Aragoncodes',
						dateSent: '2020-02-05T17:46:21.135Z',
						lastResponded: '2020-02-05T17:49:43.695Z'
					}
				]}
				fetchSurveys={jest.fn()}
			/>
		);
		expect(wrapper.debug()).toMatchSnapshot();
	});
});