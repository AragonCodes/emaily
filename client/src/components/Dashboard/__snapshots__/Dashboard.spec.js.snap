// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard should render correctly 1`] = `
"<div>
  <div>
    <div className=\\"card darken-1\\">
      <div className=\\"card-content\\">
        <span className=\\"card-title\\">
          A real campaign
        </span>
        <p>
          Did you like the service provided by Aragoncodes
        </p>
        <p className=\\"right\\">
          Sent On: 
          2/5/2020
        </p>
      </div>
      <div className=\\"card-action\\">
        <a className=\\"red-text text-accent-2\\">
          Yes: 
          1
        </a>
        <a className=\\"red-text text-accent-2\\">
          No: 
          0
        </a>
      </div>
    </div>
  </div>
  <div className=\\"fixed-action-btn\\">
    <Link to=\\"/surveys/new\\" className=\\"btn-floating pulse btn-large red accent-2\\">
      <i className=\\"material-icons\\">
        add
      </i>
    </Link>
  </div>
</div>"
`;
